indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -418,6 +419,13 @@ private TaskStatus runInternal(TaskToolbox toolbox) throws Exception
 
     try (final RecordSupplier<PartitionIdType, SequenceOffsetType, RecordType> recordSupplier = task.newTaskRecordSupplier()) {
 
+      if (toolbox.getMonitorScheduler() != null) {
+        final Monitor monitor = recordSupplier.monitor();
+        if (monitor != null) {
+          toolbox.getMonitorScheduler().addMonitor(monitor);
+        }
+      }
+
```

Comment:
Instead of registering it hear, this should really be the job of the `newTaskRecordSupplier()` call.  How about we change that method to take `TaskToolbox` and add a default implementation of that which delegates to the no-argument call.

Reply:
Sounds fine. I'm assuming the goal of the default implementation would be to generally switch to `newTaskRecordSupplier(TaskToolbox)`, but maintain compatibility with existing third-party extensions. So to do that, I updated the patch with these changes:

- Deprecate `newTaskRecordSupplier()`, but keep it around to avoid breaking extenstions.
- Add `newTaskRecordSupplier(TaskToolbox)` and switch callers and builtin extensions to use that.

LMK if this matches what you had in mind.